From a68330455be5855b8ea956d85b0efe8e8946fe4e Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Tue, 17 Mar 2026 06:34:32 -0700
Subject: [PATCH] Use coin-or-cbc from the env

---
 pulp/apis/coin_api.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/pulp/apis/coin_api.py b/pulp/apis/coin_api.py
index c69d278..fdad9ba 100644
--- a/pulp/apis/coin_api.py
+++ b/pulp/apis/coin_api.py
@@ -28,6 +28,8 @@ from __future__ import annotations
 import os
 from typing import TYPE_CHECKING
 
+import shutil
+
 from .. import constants
 from .core import (
     LpSolver,
@@ -56,9 +58,9 @@ coinMP_path = ["libCoinMP.so"]
 # workaround for (https://github.com/coin-or/pulp/issues/802)
 if operating_system == "osx":
     arch = "i64"
-pulp_cbc_path = os.path.join(
-    os.path.dirname(__file__), f"../solverdir/cbc/{operating_system}/{arch}/{cbc_path}"
-)
+pulp_cbc_path = shutil.which("cbc") or os.path.join(
+    os.path.dirname(__file__), f"../solverdir/cbc/{operating_system}/{arch}/{cbc_path}"                                                                                                     
+)   
 
 
 class COIN_CMD(LpSolver_CMD):
-- 
2.39.3 (Apple Git-146)

